log/sequencer.go

```diff
@@ -230,12 +237,57 @@ func (s Sequencer) initMerkleTreeFromStorage(ctx context.Context, currentRoot tr
 	return s.buildMerkleTreeFromStorageAtRoot(ctx, currentRoot, tx)
 }
 
-// SequenceBatch wraps up all the operations needed to take a batch of queued leaves
-// and integrate them into the tree.
-// TODO(Martin2112): Can possibly improve by deferring a function that attempts to rollback,
-// which will fail if the tx was committed. Should only do this if we can hide the details of
-// the underlying storage transactions and it doesn't create other problems.
-func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, guardWindow, maxRootDurationInterval time.Duration) (int, error) {
+// sequencingTask is used by Sequencer to extract (pre-)sequenced LogLeaf
+// entries from the storage and commit to the ordering.
+type sequencingTask interface {
+	// fetch returns a batch of sequenced entries obtained from storage. If the
+	// storage does not provide ordering, the latter is defined on the fly.
+	fetch(ctx context.Context, limit int, cutoff time.Time) ([]*trillian.LogLeaf, error)
+
+	// update makes sequencing persisted in storage, if not yet.
+	update(ctx context.Context, leaves []*trillian.LogLeaf) error
+}
+
+// logSequencingTask is a sequencingTask implementation for "normal" Log mode.
```

Comment:
Be nice if the comment indicated what the `logSequencingTask` does, rather than just what interface it implements (although that's useful information too).  Maybe something like "...for 'normal' Log mode, which assigns consecutive sequence numbers to leaves as they're read from the pending unsequenced entries"